sunpy/coordinates/frames.py

```diff
@@ -133,6 +133,10 @@ def __init__(self, *args, **kwargs):
         if not kwargs.pop('wrap_longitude', True):
             self._wrap_angle = None
 
+        # If obstime is not provided but observer has an obstime, use that as the obstime
+        if 'obstime' not in kwargs and 'observer' in kwargs and getattr(kwargs['observer'], 'obstime', None) is not None:
```

Comment:
The distinction is a bit academic/pathological, but I want to set the `obstime` on the frame only if `observer.obstime` is not `None`.  For example, although all of our frames default to `obstime=None`, we could potentially have a frame that has a non-`None` default value and is also fine with an observer coordinate that has `observer.obstime=None` (e.g., because the only relevant aspect of the observer is the heliocentric radius, a la `HeliographicCarrington`).  In that case, setting the frame's `obstime` to `None` would be the wrong thing to do.

Reply:
Ah makes sense, thanks for explaining 😄